### PR TITLE
Stavan/readme

### LINKED ISSUE #9: Initialize README
In the last week of the project, we should spend some time working on the README for this repository. In addition to explaining what this repository contains and how it can/should be used, the document should mention how future students can contribute if they want to.

### PR DESCRIPTION
**Pull request by Stavan-Jain**
Adds README

### CODE REVIEW

**Review by chenne19 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Algebra In Lean
+
+This is the repository for Algebra In Lean (AIL), an interactive problem set
+that takes users through an advanced course in Lean through the lens of abstract
+algebra. The goal of this problem set is to introduce users to formalization
+and Lean 4 while also assisting in writing rigorous proofs in
+group theory. Beginning with introductions to logic-based reasoning in Lean, each chapter takes users through various topics in group theory. Akin to Kevin
```
Comment (4rim) [inline README.md]:
```suggestion
This is the repository for Algebra In Lean (AIL), an interactive problem set
that takes users through an advanced course in Lean through the lens of
abstract algebra. The goal of this problem set is to introduce users to
formalization and Lean 4 while also assisting in writing rigorous proofs in
group theory. Beginning with introductions to logic-based reasoning in Lean,
each chapter takes users through various topics in group theory. Akin to Kevin
Buzzard’s [Formalising
Mathematics](https://www.ma.imperial.ac.uk/~buzzard/xena/formalising-mathematics-2024)
at Imperial College London, AIL includes chapters on group axioms,
homomorphisms, and subgroups, each separated into problem sheets. These problem
sheets include both the formalization of mathematical structures and exercises
on proving related theorems.
```

Just to keep wrapping (at 80, I think?) consistent. Fix "Formalizing Mathematics course" to "Formalizing Mathematics". Fix "the Imperial College London" to "Imperial College London".

File: README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+* Ricardo Prado Cunha
+
+## Credits
+
+We thank Kevin Buzzard and his
+[formalising-mathematics-2024](https://github.com/ImperialCollegeLondon/formalising-mathematics-2024?tab=readme-ov-file)
+repository for inspiring much of the infrastructure of this interactive problem
+set.
+
+We thank Dr. Colleen Robles for organizing and managing this project
```
Comment (4rim) [inline README.md]:
```suggestion
We thank Dr. Colleen Robles for organizing and managing this project.
```

Would probably be good to add a period here lol

**Review by 4rim — CHANGES_REQUESTED** [review]
Nitpicks but otherwise everything looks great!

File: README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+## Local installation
+
+First you need to install Lean 4. Instructions for doing that are
+[here](https://leanprover-community.github.io/get_started.html#regular-install).
+
+Then you will need to clone this repository from github onto your computer. Open
+up the same command line which you used to install Lean 4 and type:
+
+```bash
+git clone git@https://github.com/MAO3J1m0Op/algebra-in-lean.git
```
Comment (MithicSpirit) [inline README.md]: The `git@` is only when using SSH, which is not the case here.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+* Will Harris
+* Justin Morrill
+* Dominic King
+* Adam Kern
+* Clara Henne
+* Arim Lim
+* William Ho
+* Stavan Jain
+* Anoushka Sinha
+* Ricardo Prado Cunha
```
Comment (MithicSpirit) [inline README.md:56]: Names should be sorted alphabetically by last name (with PMs at the end)

File: README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+## Credits
+
+We thank Kevin Buzzard and his
+[formalising-mathematics-2024](https://github.com/ImperialCollegeLondon/formalising-mathematics-2024?tab=readme-ov-file)
+repository for inspiring much of the infrastructure of this interactive problem
+set.
+
+We thank Dr. Colleen Robles for organizing and managing this project
+
+We thank the Math+ program and the Duke Mathematics department for providing the
```
Comment (MithicSpirit) [inline README.md]: Maybe "We thank Duke's Math+ program and Mathematics department[]"?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+* Arim Lim
+* William Ho
+* Stavan Jain
+* Anoushka Sinha
+* Ricardo Prado Cunha
+
+## Credits
+
+We thank Kevin Buzzard and his
+[formalising-mathematics-2024](https://github.com/ImperialCollegeLondon/formalising-mathematics-2024?tab=readme-ov-file)
```
Comment (MithicSpirit) [inline README.md]: Imo remove the URL parameters. `https://github.com/ImperialCollegeLondon/formalising-mathematics-2024` is fine.

**Comment by Stavan-Jain** [comment]
> Nitpicks but otherwise everything looks great!

Addressed :)

File: README.md (hunk 1 above)
Comment (william11339) [inline README.md]: "After an introduction to" sounds better since chapter 0 isn't a topic in group theory.

**Review by MithicSpirit — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (Stavan-Jain) [reply README.md]: Should i just remove that `git@`?

**Review by william11339 — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (MithicSpirit) [reply README.md]: yeah. you can try running the command locally (in a test directory) to confirm whether it works or not.

**Review by william11339 — APPROVED** [review]

**Review by 4rim — APPROVED** [review]
lgtm!

**Review by MithicSpirit — CHANGES_REQUESTED** [review]
```patch
diff --git a/README.md b/README.md
index 1259692..21f19ba 100644
--- a/README.md
+++ b/README.md
@@ -3,27 +3,27 @@
 This is the repository for Algebra In Lean (AIL), an interactive problem set
 that takes users through an advanced course in Lean through the lens of abstract
 algebra. The goal of this problem set is to introduce users to formalization and
 Lean 4 while also assisting in writing rigorous proofs in abstract algebra.
 After an introduction to logic-based reasoning in Lean, each chapter takes users
 through various topics in group theory. Akin to Kevin Buzzard’s [Formalising
-Mathematics](https://www.ma.imperial.ac.uk/~buzzard/xena/formalising-mathematics-2024)
-at Imperial College London, AIL includes chapters on group axioms,
-homomorphisms, and subgroups, each separated into problem sheets. These problem
-sheets include both the formalization of mathematical structures and exercises
-on proving related theorems.
+Mathematics][FM] at Imperial College London, AIL includes chapters on group
+axioms, homomorphisms, and subgroups, each separated into problem sheets. These
+problem sheets include both the formalization of mathematical structures and
+exercises on proving related theorems.
 
-Also included in AIL is the [Lean
-Blueprint](https://mao3j1m0op.github.io/algebra-in-lean/blueprint/), which
-provides a dependency graph of lemmas and definitions, a top-down view of
-formalization progress, and a bottom-up view of individualized nodes. While this
-problem set presents an extensive introduction to the formalization of algebra
-and group theory mathematics, there is much left for future contributors. We
-hope that others will add to the problem set in the future, including examples
-related to quotient groups, group actions, Sylow theorems, ring theory, and much
-more.
+Also included in AIL is the [Lean Blueprint][LB], which provides a dependency
+graph of lemmas and definitions, a top-down view of formalization progress, and
+a bottom-up view of individualized nodes. While this problem set presents an
+extensive introduction to the formalization of algebra and group theory
+mathematics, there is much left for future contributors. We hope that others
+will add to the problem set in the future, including examples related to
+quotient groups, group actions, Sylow theorems, ring theory, and much more.
+
+[FM]: https://www.ma.imperial.ac.uk/~buzzard/xena/formalising-mathematics-2024
+[LB]: https://mao3j1m0op.github.io/algebra-in-lean/blueprint/
 
 ## Local installation
 
 First you need to install Lean 4. Instructions for doing that are
 [here](https://leanprover-community.github.io/get_started.html#regular-install).
 
@@ -52,17 +52,17 @@ course inside a subdirectory called `AlgebraInLean`.
 * Stavan Jain
 * Anoushka Sinha
 * Ricardo Prado Cunha
 
 ## Credits
 
-We thank Kevin Buzzard and his repository
-https://github.com/ImperialCollegeLondon/formalising-mathematics-2024
-for inspiring much of the infrastructure of this interactive problem
-set.
+We thank Kevin Buzzard and his repository for [Formalizing Mathematics][FMGH]
+for inspiring much of the infrastructure of this interactive problem set.
 
 We thank Dr. Colleen Robles for organizing and managing this project.
 
 We thank the Duke's Math+ program and Mathematics department for providing the
 opportunity and the resources to work on this project.
 
-This work was partially supported by NSF grants DMS-2003404 and ****
+This work was partially supported by NSF grants DMS-2003404 and \*\*\*\*.
+
+[FMGH]: https://github.com/ImperialCollegeLondon/formalising-mathematics-2024
```

**Review by MithicSpirit — APPROVED** [review]

**Review by MAO3J1m0Op — CHANGES_REQUESTED** [review]
[changes.patch](https://github.com/user-attachments/files/16188464/changes.patch)
Delete a colon and add a heading that mentions that we have a docs page.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+## Contributors:
+
+* Will Harris
+* Clara Henne
+* William Ho
+* Adam Kern
+* Dominic King
+* Arim Lim
+* Justin Morrill
+* Stavan Jain
```
Comment (Anoushka1406) [inline README.md:55]: not ordered by last name (at least for project managers)

**Comment by Anoushka1406** [comment]
kinda slayed overall

**Review by Anoushka1406 — APPROVED** [review]

**Review by justin-t-morrill — APPROVED** [review]

**Review by dominicking7 — APPROVED** [review]
lgtm

File: README.md (hunk 7 above)
Comment (Rebelscolonel) [inline README.md:55]: I think it makes sense to add a line ## Project Managers
before listing their names.  (Akin to ## Contributors above)

**Comment by MAO3J1m0Op** [comment]
@GitHub: this closes #9.

**Comment by Stavan-Jain** [comment]
I believe this is ready for merge into master

File: README.md (hunk 7 above)
Comment (MithicSpirit) [reply README.md:55]: Stavan should go first since the last name is "Prado Cunha" not just "Cunha"

File: README.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+https://github.com/ImperialCollegeLondon/formalising-mathematics-2024
+for inspiring much of the infrastructure of this interactive problem
+set.
+
+We thank Dr. Colleen Robles for organizing and managing this project.
+
+We thank the Duke's Math+ program and Mathematics department for providing the
+opportunity and the resources to work on this project.
+
+This work was partially supported by NSF grants DMS-2003404 and ****
```
Comment (MAO3J1m0Op) [inline README.md]: Do we know the second NSF grant number?

Comment (MithicSpirit) [reply README.md]: Not yet. We can ask them today. Maybe remove that part and add it in when we know? Or just wait to merge this until then.

Comment (Stavan-Jain) [reply README.md]: I think we should merge before the presentation. I'll remove the **** if that's better

Comment (MithicSpirit) [reply README.md]: We could ask them during lunch before we present.

**Review by MAO3J1m0Op — APPROVED** [review]
Deleted NSF grant line; see latest commit. We will add it later once Prof. Hahn sends us all the information. Otherwise perfect and ready for merge.

**Pull request merged**